src/mlpack/methods/bayesian_ridge/bayesian_ridge.hpp

```diff
@@ -0,0 +1,298 @@
+/**
+ * @file bayesian_ridge.hpp
+ * @author Clement Mercier
+ *
+ * Definition of the BayesianRidge class, which performs the
+ * bayesian linear regression. According to the armadillo standards,
+ * all the functions consider data in column-major format.
+**/
+#ifndef MLPACK_METHODS_BAYESIAN_RIDGE_BAYESIAN_RIDGE_HPP
+#define  MLPACK_METHODS_BAYESIAN_RIDGE_BAYESIAN_RIDGE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack{
+namespace regression{
+/**
+ * This class implements the bayesian linear regression. "Bayesian treatment
+ * of linear regression, which will avoid the over-fitting problem of maximum
+ * likelihood, and which will also lead to automatic methods of determining
+ * model complexity using the training data alone.", C.Bishop.
```

Comment:
Maybe it's better to not quote directly from Bishop but instead describe the method a little bit?

"Bayesian linear regression is an adaptation of linear regression that ... "

Reply:
I wrote a new description. Let me know if if it is still not satisfactory.

Reply:
Looks great; thank you! :+1: